js/main.js

```diff
@@ -0,0 +1,21 @@
+var $photoUrlInput = document.querySelector('#photourl');
+var $img = document.querySelector('img');
+var $form = document.querySelector('form');
+
+$photoUrlInput.addEventListener('input', function (event) {
+  $img.setAttribute('src', event.target.value);
+});
+
+$form.addEventListener('submit', function (event) {
+  event.preventDefault();
+  var $formInfo = {
+    title: $form.elements.title.value,
+    photoUrl: $form.elements.photourl.value,
+    notes: $form.elements.textarea.value,
+    entryId: data.nextEntryId
+  };
+  data.nextEntryId++;
+  data.entries.push($formInfo);
```

Comment:
You should use `unshift()` here, so that the newest entry is always at the start of the array. This will come in handy in the next step when you want to render the entries